df_py/predictoor/test/test_predictoor_csvs.py

```diff
@@ -47,3 +47,35 @@ def test_predictoordata(tmp_path):
             loaded_predictoor.correct_prediction_count
             == original_predictoor.correct_prediction_count
         )
+
+
+@enforce_types
+def test_predictoorrewards(tmp_path):
+    target_csv = """predictoor_addr,reward
+0x0000000000000000000000000000000000000000,10.0
+0x1000000000000000000000000000000000000000,20.0
+0x2000000000000000000000000000000000000000,30.0
+0x3000000000000000000000000000000000000000,40.0
+0x4000000000000000000000000000000000000000,50.0"""
+
+    # generate random rewards
+    predictoor_rewards = {}
+    for i in range(5):
+        predictoor_rewards[f"0x{i}000000000000000000000000000000000000000"] = (
+            i + 1
+        ) * 10.0
+
+    csv_dir = str(tmp_path)
+    csvs.save_predictoor_rewards_csv(predictoor_rewards, csv_dir)
+
+    with open(csvs.predictoor_rewards_csv_filename(csv_dir), "r") as loaded_data:
+        data = loaded_data.read().strip()
```

Comment:
here is an instance where csv dictreader and dictwriter would have helped. The test data could simply be a dictionary, not a string and loaded like this